### PR TITLE
refactor: consolidate MMKV usage

### LINKED ISSUE #2161: refactor: use canonical MMKV instance
Currently we initialize the MMKV in two places in the app:
- https://github.com/techmatters/terraso-mobile-client/blob/f197e61cd11310a06dafe75ef6ae5862a076ee23/dev-client/src/hooks/useStorage.ts#L20
- https://github.com/techmatters/terraso-mobile-client/blob/22d8653f1dc075b524f188789606615074f142ba/dev-client/src/config/index.ts#L27

The MMKV docs recommend initializing only once, and we use separate settings in both instances. We should define a single place where MMKV is loaded.

We also need to understand how the `withEncryption` functionality works, so as part of completing this work we should document that alongside the initialization.

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Consolidates our usage of MMKV.

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Fixes #2161

### Verification steps
Everything should work the same.

### CODE REVIEW

**Review by tm-ruxandra — APPROVED** [review]

File: dev-client/src/config/index.ts
Comment (shrouxm) [inline dev-client/src/config/index.ts:86]: note: i moved this to `featureFlags.ts` because:
- we already expose a function which reads off these `APP_CONFIG` values and it seems better to not have more than one way to do that
- it makes it easier to test `featureFlags.ts` as an isolated module

**Comment by shrouxm** [comment]
waiting for feedback on tests before merging

File: dev-client/src/config/index.ts
Comment (knipec) [reply dev-client/src/config/index.ts:86]: Yay, I like it!

File: dev-client/src/persistence/kvStorage.ts
Comment (knipec) [inline dev-client/src/persistence/kvStorage.ts:41]: Question: I assume it will flush on app close -- is preventing data loss if the app crashes the main impetus for considering flushing manually? Are there other use cases you have in mind?

**Review by knipec — APPROVED** [review]

File: dev-client/package-lock.json
Comment (paulschreiber) [inline dev-client/package-lock.json:101]: issue: you removed this file by accident.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/jest/unit/setup.ts
Comment (tm-ruxandra) [inline dev-client/jest/unit/setup.ts:21]: I don't know if I agree with making this a globally-configured setting in our unit tests; is it possible to just set this up for the kv storage and feature flag test files? The logic seems so minimal and we don't know what the cost is of clearing the mmkv mock for every test case.

File: dev-client/package-lock.json
Comment (shrouxm) [reply dev-client/package-lock.json:101]: not an accident, it's no longer used

File: dev-client/src/persistence/kvStorage.ts
Comment (shrouxm) [reply dev-client/src/persistence/kvStorage.ts:41]: that's what i was thinking about!

File: dev-client/package-lock.json
Comment (paulschreiber) [reply dev-client/package-lock.json:101]: sorry for the mixup. GitHub's UI made it looked like to remove the whole file.

**Review by paulschreiber — APPROVED** [review]

File: dev-client/jest/unit/setup.ts
Comment (shrouxm) [reply dev-client/jest/unit/setup.ts:21]: i looked into it a bit, it looks like the cost of clearing the mock is more or less setting a handful of pointers to null and leaving some objects to be GC'd

it is possible to do it in individual test files, but it can be awkward because you need to manually make sure the imports resolve in the right order since the mock expects to be required in a setup file like this before any of the library's use sites are resolved

if it turns out to be a problem it should be straightforward to manually inline it though

**Pull request merged**
